packages/css/src/identifier.ts

```diff
@@ -3,23 +3,46 @@ import hash from '@emotion/hash';
 import { getIdentOption } from './adapter';
 import { getAndIncrementRefCounter, getFileScope } from './fileScope';
 
-function getDevPrefix(debugId: string | undefined) {
+function getDevPrefix({
+  debugId,
+  debugFileName,
+}: {
+  debugId?: string;
+  debugFileName: boolean;
+}) {
   const parts = debugId ? [debugId.replace(/\s/g, '_')] : [];
-  const { filePath } = getFileScope();
 
-  const matches = filePath.match(
-    /(?<dir>[^\/\\]*)?[\/\\]?(?<file>[^\/\\]*)\.css\.(ts|js|tsx|jsx)$/,
-  );
+  if (debugFileName) {
+    const { filePath } = getFileScope();
+
+    const matches = filePath.match(
+      /(?<dir>[^\/\\]*)?[\/\\]?(?<file>[^\/\\]*)\.css\.(ts|js|tsx|jsx)$/,
+    );
 
-  if (matches && matches.groups) {
-    const { dir, file } = matches.groups;
-    parts.unshift(file && file !== 'index' ? file : dir);
+    if (matches && matches.groups) {
+      const { dir, file } = matches.groups;
+      parts.unshift(file && file !== 'index' ? file : dir);
+    }
   }
 
   return parts.join('_');
 }
 
-export function generateIdentifier(debugId: string | undefined) {
+interface GenerateIdentifierOptions {
+  debugId?: string;
+  debugFileName?: boolean;
+}
+
+export function generateIdentifier(debugId?: string): string;
+export function generateIdentifier(options?: GenerateIdentifierOptions): string;
```

Comment:
I think you don't really need the overloads here as it always the same number of params. It's not wrong though. Upto you.

Reply:
True, but it's partly due to naming when consumers inspect the type, otherwise the type of `options` is also `string` which doesn't tell you it's a debug ID.

Reply:
Good point, happy to leave it then